apps/cow-tools/src/app/milkman/(components)/OrdersTable.tsx

```diff
@@ -53,38 +60,211 @@ export function OrderTable({ orders }: { orders: AllSwapsQuery["swaps"] }) {
-          <Table.BodyCell>5</Table.BodyCell>
-          <Table.BodyCell>Test 2</Table.BodyCell>
-          <Table.BodyCell>
-            {TransactionStatus.CANCELATION_TO_BE_EXECUTED}
-          </Table.BodyCell>
-          <Table.BodyCell>
-            <CancelButton
-              status={TransactionStatus.CANCELATION_TO_BE_EXECUTED}
-            />
-          </Table.BodyCell>
-        </Table.BodyRow>
+        {transactions?.length === 0 && (
+          <Table.BodyRow>
+            <Table.BodyCell colSpan={6}>
+              <h1 className="text-md text-slate12 m-2 text-center w-full">
+                This address didn't made any order on Milkman yet
+              </h1>
+            </Table.BodyCell>
+          </Table.BodyRow>
+        )}
       </Table.Body>
     </Table>
   );
 }
 
-function TableRow({ order }: { order: AllSwapsQuery["swaps"][0] }) {
-  const transactionStatus = TransactionStatus.MILKMAN_CREATED;
+function TableRowTransaction({
+  transaction,
+  hasToken,
+  cowOrders,
+}: {
+  transaction: AllTransactionFromUserQuery["users"][0]["transactions"][0];
+  hasToken?: boolean[];
+  cowOrders?: ICowOrder[][];
+}) {
+  function getSwapStatus(hasToken?: boolean, cowOrders?: ICowOrder[]) {
+    if (!cowOrders || hasToken === undefined) {
+      return SwapStatus.MILKMAN_CREATED;
+    }
+
+    const anyOrderWasExecuted = cowOrders.some(
+      (order) => order.status == "fulfilled",
+    );
+
+    if (anyOrderWasExecuted) {
+      return SwapStatus.EXECUTED;
+    }
+
+    if (!anyOrderWasExecuted && !hasToken) {
+      return SwapStatus.CANCELED;
+    }
+
+    if (cowOrders.length > 0 && hasToken) {
+      return SwapStatus.ORDER_PLACED;
+    }
+
+    return SwapStatus.MILKMAN_CREATED;
+  }
+
+  function getTransactionStatus(swapStatus: SwapStatus[]) {
+    if (swapStatus.every((status) => status === SwapStatus.CANCELED)) {
+      return TransactionStatus.CANCELED;
+    }
+    if (swapStatus.every((status) => status === SwapStatus.EXECUTED)) {
+      return TransactionStatus.EXECUTED;
+    }
+
+    if (
+      swapStatus.every(
+        (status) =>
+          status === SwapStatus.CANCELED || status === SwapStatus.EXECUTED,
+      )
+    ) {
+      return TransactionStatus.EXECUTED_AND_CANCELED;
+    }
+    if (swapStatus.some((status) => status === SwapStatus.EXECUTED)) {
+      return TransactionStatus.PARTIALLY_EXECUTED;
+    }
+
+    const allOrdersPlaced = swapStatus.every(
+      (status) => status === SwapStatus.ORDER_PLACED,
+    );
+
+    if (allOrdersPlaced) {
+      return TransactionStatus.ORDER_PLACED;
+    }
+
+    return TransactionStatus.MILKMAN_CREATED;
+  }
+
+  const swapStatus = transaction.swaps.map((swap, i) =>
+    getSwapStatus(hasToken?.[i], cowOrders?.[i]),
+  );
+  const transactionStatus = getTransactionStatus(swapStatus);
   const txUrl = buildBlockExplorerTxUrl({
-    chainId: order.chainId,
-    txHash: order.transactionHash,
+    chainId: transaction.swaps[0].chainId,
```

Comment:
this is because there is no way we can have a multichain view, right? maybe leave a comment for the future

Reply:
The app is connected to just your wallet (one chain), so I think that a multichain view doesn't make sense in this case. But let me know your suggestions